src/Sylius/Component/Core/Dashboard/SalesDataProvider.php

```diff
@@ -29,35 +29,41 @@ public function __construct(EntityManagerInterface $entityManager)
         $this->entityManager = $entityManager;
     }
 
-    public function getLastYearSalesSummary(ChannelInterface $channel): SalesSummaryInterface
-    {
-        $startDate = (new \DateTime('first day of next month last year'))->format('Y/m/d');
-        $endDate = (new \DateTime('last day of this month'))->format('Y/m/d');
+    public function getSalesSummary(
+        \DateTimeInterface $startDate,
+        \DateTimeInterface $endDate,
+        string $interval,
+        ChannelInterface $channel,
+        string $dateFormat
+    ): SalesSummaryInterface {
+        $formattedStartDate = $startDate->format('Y/m/d H:i:s');
+        $formattedEndDate = $endDate->format('Y/m/d H:i:s');
         $channelId = $channel->getId();
 
         $query = $this->entityManager->getConnection()->query(
-            "SELECT
-                DATE_FORMAT(checkout_completed_at, '%m.%y') AS \"date\",
-                SUM(total) as \"total\"
+            "
+            SELECT
+                DATE_FORMAT(checkout_completed_at, '%m.%y') AS date,
+                DATE_FORMAT(checkout_completed_at, '%y.%m') AS month,
+                DATE_FORMAT(checkout_completed_at, '%y.%m.%d') AS day,
+                DATE_FORMAT(checkout_completed_at, '%y.%m.%d %H') AS hour,
+                DATE_FORMAT(checkout_completed_at, '%y') AS year,
+            SUM(total) as total
             FROM sylius_order
             WHERE (channel_id = $channelId)
-            AND (checkout_completed_at BETWEEN '$startDate' AND '$endDate')
+            AND (checkout_completed_at BETWEEN '$formattedStartDate' AND '$formattedEndDate')
             AND (payment_state = 'paid')
-            GROUP BY date;"
+            GROUP BY '$interval', checkout_completed_at;"
```

Comment:
`$interval` and other injected variables directly to this SQL query should be replaced with prepared statements to mitigate SQL injection risks.

Reply:
There is a refactor proposed already in #11405. We should keep original commits if only possible